src/OpenLoco/Entities/EntityManager.cpp

```diff
@@ -199,7 +199,7 @@ namespace OpenLoco::EntityManager
         return removeFromSpatialIndex(entity, index);
     }
 
-    void moveSpatialEntry(EntityBase& entity, const Map::Pos2& loc)
+    void moveSpatialEntry(EntityBase& entity, const Map::Pos3& loc)
```

Comment:
Previously if the caller passed the loc by reference and the reference was the entities position you would end up zeroing out the z coordinate. Solution is to either not pass by reference or to always pass the full size of the coordinate despite only using the first 2 coordinates.